### PR TITLE
Update solidity version to ^0.8.4

### PR DESCRIPTION
**Pull request by ScreamingHawk**
#54

### CODE REVIEW

File: src/contracts/utils/WrapAndNiftyswap.sol [hunk 1]
```diff
@@ -1,163 +1,149 @@
...
+    function wrapAndSwap(uint256 _maxAmount, address _recipient, bytes calldata _niftyswapOrder) external {
+        // Decode niftyswap order
+        INiftyswapExchange.BuyTokensObj memory obj;
+        (, obj) = abi.decode(_niftyswapOrder, (bytes4, INiftyswapExchange.BuyTokensObj));
+
+        // Force the recipient to not be set, otherwise wrapped token refunded will be
+        // sent to the user and we won't be able to unwrap it.
+        require(
+            obj.recipient == address(0x0) || obj.recipient == address(this),
+            "WrapAndNiftyswap#wrapAndSwap: ORDER RECIPIENT MUST BE THIS CONTRACT"
```
Comment (acrylix) [inline src/contracts/utils/WrapAndNiftyswap.sol:66]: maybe we can take this chance to rename all revert messages similar to 

https://github.com/0xsequence/niftyswap/blob/be69ffd8576bb25b1a234b13eacdddeff1114d4a/src/contracts/exchange/NiftyswapExchange20.sol#L146

**Review by acrylix — APPROVED** [review]

File: src/contracts/exchange/NiftyswapFactory.sol [hunk 2]
```diff
@@ -1,39 +1,44 @@
...
-
-    // Emit event
-    emit NewExchange(_token, _currency, _currencyID, address(exchange));
-  }
-
+    /**
+     * |
+     * |       Events And Variables        |
+     * |__________________________________
+     */
```
[the remark this answers is not in the document]
Comment (ScreamingHawk) [reply src/contracts/exchange/NiftyswapFactory.sol:12]: Formatter broke them... I'll change the layout to something the formatter like

File: src/contracts/utils/WrapAndNiftyswap.sol (hunk 1 above)
Comment (acrylix) [reply src/contracts/utils/WrapAndNiftyswap.sol:66]: was originally done for contract code size limit

**Review by PhABC — APPROVED** [review]

File: src/contracts/utils/WrapAndNiftyswap.sol (hunk 1 above)
Comment (ScreamingHawk) [reply src/contracts/utils/WrapAndNiftyswap.sol:66]: Ok, I'll only update to `NE#XX` for now

**Pull request merged**